src/tools/wasp_sqlite.cc

```diff
@@ -0,0 +1,564 @@
+         static_cast<int>(global.value.global_type.valtype),
+         static_cast<int>(global.value.global_type.mut));
+
+    InsertConstantExpression(global.value.init, "global_init", global.index);
+  }
+}
+
+void Tool::DoExportSection(LazyExportSection section) {
+  for (auto export_ : enumerate(section.sequence)) {
+    Exec("insert into export values ({}, {}, \"{}\", {});", export_.index,
+         static_cast<int>(export_.value.kind), export_.value.name,
+         export_.value.index);
+  }
+}
+
+void Tool::DoStartSection(StartSection section) {
+  if (section) {
+    Exec("insert into start values ({});", section->func_index);
+  }
+}
+
+void Tool::DoElementSection(LazyElementSection section) {
+  for (auto segment : enumerate(section.sequence)) {
+    if (segment.value.is_active()) {
+      const auto& active = segment.value.active();
+      Exec("insert into element values ({}, {}, {});", segment.index,
+           static_cast<int>(segment.value.segment_type()), active.table_index);
+
+      InsertConstantExpression(active.offset, "element_offset", segment.index);
+      for (auto func : enumerate(active.init)) {
+        Exec("insert into element_init values ({}, {}, null, {});",
+             segment.index, func.index, func.value);
+      }
+    } else {
+      const auto& passive = segment.value.passive();
+      Exec("insert into element values ({}, {}, null);", segment.index,
+           static_cast<int>(segment.value.segment_type()));
+      for (auto expr : enumerate(passive.init)) {
+        auto instr = expr.value.instruction;
+        auto opcode_val = static_cast<int>(instr.opcode);
+        switch (instr.opcode) {
+          case Opcode::RefNull:
+            Exec("insert into element_init values ({}, {}, {}, null);",
+                 segment.index, expr.index, opcode_val);
+            break;
+
+          case Opcode::RefFunc:
+            Exec("insert into element_init values ({}, {}, {}, {});",
+                 segment.index, expr.index, opcode_val,
+                 instr.index_immediate());
+            break;
+
+          default:
+            break;
+        }
+      }
+    }
+  }
+}
+
+void Tool::DoCodeSection(LazyCodeSection section) {
+  sqlite3_stmt* stmt;
+  if (sqlite3_prepare_v2(
+          db, "insert into instruction values (?1, ?2, ?3, ?4, ?5, ?6);", -1,
+          &stmt, nullptr) != SQLITE_OK) {
+    print("Error: {}\n", sqlite3_errmsg(db));
+    return;
+  }
+
+  for (auto code : enumerate(section.sequence, imported_function_count)) {
+    Exec("insert into code values ({}, {}, {});", code.index,
+         file_offset(code.value.body.data), code.value.body.data.size());
+
+    for (const auto& locals : enumerate(code.value.locals)) {
+      Exec("insert into locals values ({}, {}, {}, {});", code.index,
+           locals.index, locals.value.count,
+           static_cast<int>(locals.value.type));
+    }
+
+    SpanU8 last_data = code.value.body.data;
+    Index index = 0;
```

Comment:
Can this be done with
```c++
for (auto pair : enumerate(instrs, 0) {
  const auto& instr = pair.value;
  Index index = pair.index;
```

... or I guess not because we use `it` to compute the size of the instruction

Reply:
Yeah, normally we don't need anything off of the iterator, but in this case it's useful.